### ISSUE TITLE
Extant issues with RPA client-side support in Electron Cash

### ISSUE THREAD

**Issue by cculianu** [Electron-Cash/Electron-Cash#2801]
I'm putting this here just to have it down and recorded:

- [ ] We need to set the testnet start heights (in `networks.py`) to something more recent. Right now is hard-coded to `0`. This means on old testnets like testnet3, we are having users scan over a million blocks to become up-to-date for RPA. Not good.
- [ ] When restoring a wallet from seed, it spends its time in the background scouring from height `825000`, looking for relevant txns. This process can take from tens of seconds to minutes.  Need a UI indication that the wallet is "Syncrhonizing" otherwise users may feel worried when they restore and see an empty wallet (until it finds txns).
- [ ] When creating a new wallet, right now the wallet will always use the hard-coded height of `825000` on mainnet and scan from then to present. This hardly makes sense if you *know* the wallet cannot have txns since it's new. Need to make the base_wizard try and record the height when the wallet was created or some other flag to tell the rpa_manager to not synch from beginning of time
- [ ] The rpa_manager has no concept of reorgs and cannot deal with them. It assumes heights are unique and idempotent representations of blocks.. which they are not. Heights also can go backward, etc. This is a corner case but needs to be handled... at least eventually.

**Comment by cculianu** [comment]
Ah, good idea. Not sure about the literal accuracy of that formula.. there are no variables there so it evaluates to a constant! I presume you want to parameterize it with some variable `t` which represents the approximate start date the user had in mind?

Regardless of the details of the formula I get what you are saying and that's a darned good idea tbh.

**Comment by cculianu** [comment]
I pushed a commit to at least give the UI some indication that RPA is synching in the BG.

**Comment by imaginaryusername** [comment]
There was a variable there... i fucked up by putting them in the wrong brackets so github hid it. Edited it back into existence :sweat_drops:

**Comment by imaginaryusername** [comment]
I added the +40000 to actually undershoot the overmining from genesis... but i guess we can use asert too. something like 

that would work out to be 

max(825000, ((entered_date_in_unix_time - 1605449609) / 600 - 200) 

where 1605443067 is timestamp of asert anchor, and the -200 is there to be safe.

**Comment by cculianu** [comment]
Yeah I did basically this in recent commits (except it reads asert timestamps and anchor blocks from net config).

**Comment by cculianu** [comment]
@imaginaryusername @fyookball I just pushed a commit to:

- If creating a new wallet from a new seed, forces the RPA height to be "1 day ago" (paranoia to be safe)
- If restoring an RPA wallet from seed -- prompts the user to input a date using a QDateTime dialog widget. The input field defaults to 2 weeks ago, but they can scroll back to any calendar date approx as far back as late Dec. 2023.

**Comment by imaginaryusername** [comment]
That sounds exactly like what I'd imagine how it should work. Thanks a lot!!!!
